### PR TITLE
ci: run builds on 'next' branch of git.git

### PR DESCRIPTION
**Pull request by ttaylorr**
This pull request teaches CircleCI and Travis how to run CI with the version of Git present at the tip of branch 'next'.

This is useful for adding the 'delay' capability to `git lfs filter-process` (see: https://github.com/git-lfs/git-lfs/issues/2466), since the 'delay' capability is not yet in master, or a released version of Git. This will give us a chance to test the 'delay' functionality out in CI before it's released in Git.

Once Git releases a version that contains https://github.com/git/git/commit/2841e8f81cb2820024804b9341577be1d0ce1240, we can revert this pull request.

This is required work for: https://github.com/git-lfs/git-lfs/issues/2466.

---

/cc @git-lfs/core 
/cc https://github.com/git-lfs/git-lfs/issues/2466

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ machine:
     GIT_SOURCE_REPO: https://github.com/git/git.git
     GIT_EARLIEST_SUPPORTED_VERSION: v2.0.0
     GIT_LATEST_SOURCE_BRANCH: master
+    GIT_NEXT_SOURCE_BRANCH: master
```
Comment (larsxschneider) [inline circle.yml]: Shouldn't that be `next`?

Comment (ttaylorr) [inline circle.yml]: @larsxschneider great catch! I fixed that up in: 
https://github.com/git-lfs/git-lfs/pull/2470/commits/a2e6e2d21cb904e09963715693fa64dc367c17e1.

File: .travis.yml [hunk 2]
```diff
@@ -12,6 +12,7 @@ env:
     - GIT_SOURCE_REPO="https://github.com/git/git.git"
     - GIT_EARLIEST_SUPPORTED_VERSION="v2.0.0
     - GIT_LATEST_SOURCE_BRANCH="master"
+    - GIT_NEXT_SOURCE_BRANCH="next"
```
Comment (larsxschneider) [inline .travis.yml]: You could also defined `GIT_LATEST_SOURCE_BRANCH="next"` for the time being.
That's why I called it `LATEST` and not `MASTER` back then 😄

File: circle.yml [hunk 3]
```diff
@@ -4,6 +4,7 @@ machine:
     GIT_SOURCE_REPO: https://github.com/git/git.git
     GIT_EARLIEST_SUPPORTED_VERSION: v2.0.0
     GIT_LATEST_SOURCE_BRANCH: master
+    GIT_NEXT_SOURCE_BRANCH: next
```
Comment (larsxschneider) [inline circle.yml]: Same here

File: .travis.yml (hunk 2 above)
Comment (ttaylorr) [inline .travis.yml]: I think running on `master` & `next` for the time being will be fine.

File: circle.yml (hunk 3 above)
Comment (ttaylorr) [inline circle.yml]: Is this the same line as https://github.com/git-lfs/git-lfs/pull/2470#discussion_r130989750?

**Pull request merged**